force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -213,9 +284,19 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
 
-        // Subtract the difference in years
+        // Check if either date is null
+        if (d1 == null || d2 == null) {
+        return null;
+        }
 
-        return null; // Replace null with the variable you used to store the result
+        Integer yearFirstDate = d1.year();  // Extract Year from Date 1
+        Integer yearSecondDate = d2.year(); // Extract Year from Date 2
+
+        // Subtract the difference in years
+        Integer yearsDifference = Math.abs(yearSecondDate - yearFirstDate);
+            
+            return yearsDifference;   
+           
```

Comment:
Ask yourself I am I creating a variable for no reason but to return it on the next line. If you are then just return directly. 


  return Math.abs(d1.year() - d2.year());